### PR TITLE
Issue 1971 - refactor topic types/risk categories into arrays

### LINKED ISSUE #1971: Store topic types and categories as array of strings
### Description
Store topic types and categories as array of strings instead of key/val pairs

**DB Migration:**
`mongo <host:port> -u <user> -p <password> --authenticationDatabase=admin scripts\db_migrations\v4.6\changeIssueTypesAndRiskCategoriesDatatype.js`

### Tasks
- [x] Backend udpate
- [x] Frontend update
- [x] Migration script
   - [x] Rename everything in teamspace settings
   - [x] Rename everything in comments

### PR DESCRIPTION
**Pull request by carmenfan**
This fixes #1971

**!!! This has #1966 merged in it needs to wait until #1966 is in staging. !!!**

Base branch is set to #1966 for an easier view for code review

#### Description
- DB script to convert risk categories/issue types [object] => [string]
- altered type checks to suit on the backend
- altered front end to understand the new type

#### Test cases
NOTE: subscriptionTest DB is migrated to this new format if you don't want to ruin your current db. (It has a very obvious password!)
- API should now take in [string] instead of [object] for issueTypes/risk cat
-  data should show correctly in risk/issues
-  drop down menu should show correctly in issues/risks details and new issues/risks
- search filters should function correctly
- kanban grouping should work

#### Notes:
- Auto test not updated as they only exist in #1963

### CODE REVIEW

File: backend/models/teamspaceSetting.js [hunk 1]
```diff
@@ -129,37 +129,23 @@ class TeamspaceSettings {
...
-									value,
-									label: label.trim()
-								};
+					const arrayUpdated = [];
+					const foundKeys = {};
+					for (let i = 0; i < data[key].length; ++i) {
+						const entry = data[key][i];
+						if(utils.isString(entry) && entry !== "") {
+							const value = entry.trim();
+							if (utils.hasField(foundKeys, value.toLower())) {
```
Comment (Charence) [inline backend/models/teamspaceSetting.js:138]: I get the error: `TypeError: value.toLower is not a function`

File: backend/models/teamspaceSetting.js [hunk 2]
```diff
@@ -129,37 +129,23 @@ class TeamspaceSettings {
...
+								throw responseCodes.DUPLICATED_ENTRIES;
 							}
+							foundKeys[value.toLower()] = 1;
```
Comment (Charence) [inline backend/models/teamspaceSetting.js:141]: I get the error: `TypeError: value.toLower is not a function`

File: frontend/constants/issues.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ export const PRIORITIES = {
 // FIXME: first item should be taken in topicTypes should be taken
 // for default value
 const TOPIC_TYPES = {
-	FOR_INFORMATION: 'for_information'
+	FOR_INFORMATION: 'For Information'
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply frontend/constants/issues.ts:35]: It's been address months ago I think.

https://github.com/3drepo/3drepo.io/pull/1978/files#diff-c9def28852214f55a5794dbda6ae3e28R576

File: backend/models/teamspaceSetting.js [hunk 4]
```diff
@@ -129,37 +129,23 @@ class TeamspaceSettings {
...
-									label: label.trim()
-								};
+					const arrayUpdated = [];
+					const foundKeys = {};
+					for (let i = 0; i < data[key].length; ++i) {
+						const entry = data[key][i];
+						if(utils.isString(entry) && entry !== "") {
+							const value = entry.trim();
+							if (utils.hasField(foundKeys, value.toLowerCase())) {
+								throw responseCodes.DUPLICATED_ENTRIES;
```
Comment (Charence) [inline backend/models/teamspaceSetting.js:139]: It seems like this error isn't thrown e.g. if topicTypes "Unknown" and "unknown" exist.

I also get this logged:

```
(node:6028) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'name' of undefined
    at Utils.mongoErrorToResCode (/Users/charencewong/Workspace/3drepo.io/backend/utils.js:146:10)
    at Object.responseCodes.respond (/Users/charencewong/Workspace/3drepo.io/backend/response_codes.js:397:19)
    at TeamspaceSettings.update.then.catch.err (/Users/charencewong/Workspace/3drepo.io/backend/routes/teamspace.js:523:18)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:6028) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
```

File: frontend/constants/issues.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (carmenfan) [reply frontend/constants/issues.ts:35]: I'm not sure why we need to have a topic type object at all, so i'm just removing it.

Setting issues as `For Information` is default where possible was a requirement when we did issue so we only take the first item as default value if we can't find it. So I think as it stands it's ok!

File: backend/models/teamspaceSetting.js (hunk 4 above)
Comment (carmenfan) [reply backend/models/teamspaceSetting.js:139]: fixed

**Comment by carmenfan** [comment]
@Charence All fixed now!

**Pull request merged**